main/src/com/google/refine/commands/lang/LoadLanguageCommand.java

```diff
@@ -79,32 +79,52 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             modname = "core";
         }
 
+        // Suggested languages from request, if given...
         String[] langs = request.getParameterValues("lang");
-        if (langs == null || "".equals(langs[0])) {
-            PreferenceStore ps = ProjectManager.singleton.getPreferenceStore();
-            if (ps != null) {
-                langs = new String[] { (String) ps.get("userLang") };
+        if (langs == null) {
+            langs = new String[] {};
+        }
+
+        // Always replace suggested with user preference language, if available...
+        PreferenceStore ps = ProjectManager.singleton.getPreferenceStore();
+        if (ps != null) {
+            String strLang = (String) ps.get("userLang");
+            // If user preference language exists...
+            if (!(strLang == null || strLang.isEmpty())) {
+
+                // CORRECTOR...
+                // TODO: This code may be removed sometime after the 3.7 release has been circulated.
```

Comment:
The logic looks good to me. Although the code under the TODO should probably be kept in case people wait until a version later than 3.7 to update.